src/components/SignerPanel/SignerPanelContent.js

```diff
@@ -62,28 +65,45 @@ class ActionPathsContent extends React.Component {
       </span>
     )
   }
-  render() {
-    const { intent: { description, to, tx }, paths } = this.props
-    const { selected } = this.state
-    const radioItems = paths.map(({ appName, description }) => {
+  getPathRadioItem(path) {
+    // Slice off the intention (last transaction in the path)
+    path = path.slice(0, path.length - 1)
+
+    const title = path.map(({ name }, index) => {
```

Comment:
I think this `.map()` is meant to produce a string? Might be better as a `.reduce()` (even more so as a `.reverse().reduce()`; it'll avoid the last index calculations).

Reply:
It's meant to produce an array of React components since we want the title of the app if it's shortened.

Also, how does reduce avoid the last index calculation?

Reply:
Ahh, I see, it's a list of `<span />`s.

Reply:
> Also, how does reduce avoid the last index calculation?

Ended up finding `.map().join()` was a better fit, but you can do something similar with reduces (the `.reverse()` may be useful sometimes when it's easier to fold from the right side):

```js
const str = [str1, str2, str3].reduce((str, nextStr) => str ? `${str} -> ${nextStr}` : nextStr, '')
```